### PR TITLE
Add Sparse Support for Frac Operator

### PR DESCRIPTION
**Pull request by hlahkar**

### CODE REVIEW

**Comment by mruberry** [comment]
Hey @hlahkar! Thanks for this implementation. We should wait until we can test this using OpInfos to greatly simplify the testing, and we should also look at creating an easier way to implement unary pwise operators for sparse tensors.

Take a look at how most unary ops are implemented here, for example:

https://github.com/pytorch/pytorch/blob/39a10fb6528b355791993a67b895934e3c902765/aten/src/ATen/native/UnaryOps.cpp#L253

I would think that we can create a mechanism just as simple for supporting sparse tensors with these pwise ops, too?

**Comment by mruberry** [comment]
It doesn't seem like we use the that pattern in sparse, but for unary pwise ops I don't think we need it. For example with asin we have this device-generic dispatch:

https://github.com/pytorch/pytorch/blob/c0deb231db76dbea8a9d326401417f7d1ce96ed5/aten/src/ATen/native/native_functions.yaml#L701

And it redispatches to strided asin:

https://github.com/pytorch/pytorch/blob/c892c3ac9affb428d28696270f337007729ed914/aten/src/ATen/native/sparse/SparseTensorMath.cpp#L162

So the stub logic is handled by the existing unary ops.

**Comment by hlahkar** [comment]
> It doesn't seem like we use the that pattern in sparse, but for unary pwise ops I don't think we need it. For example with asin we have this device-generic dispatch:
> 
> https://github.com/pytorch/pytorch/blob/c0deb231db76dbea8a9d326401417f7d1ce96ed5/aten/src/ATen/native/native_functions.yaml#L701
> 
> And it redispatches to strided asin:
> 
> https://github.com/pytorch/pytorch/blob/c892c3ac9affb428d28696270f337007729ed914/aten/src/ATen/native/sparse/SparseTensorMath.cpp#L162
> 
> So the stub logic is handled by the existing unary ops.

@mruberry - Thanks for the response. I am following a similar pattern as followed in asin

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -1059,6 +1059,27 @@ def sample_inputs_pinverse(op_info, device, dtype, requires_grad=False):
            supports_tensor_out=False,
            sample_inputs_func=sample_inputs_pinverse,
            decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack]),
+    UnaryUfuncInfo('frac',
+                   ref=np.arcsin,
```
Comment (cpuhrsch) [inline torch/testing/_internal/common_methods_invocations.py:1063]: Why is the reference arcsin?

**Comment by cpuhrsch** [comment]
Overall this seems reasonable to me. @hlahkar could you also rebase please? There's a conflict.

Further, could you add a sentence to the PR description on what you need this for?

Could you also please follow the pattern used in https://github.com/pytorch/pytorch/pull/69292 ?

**Pull request closed without merging**